e2e/test/prerequisites/E2ETestsSetup/LoadEnvironmentVariablesFromKeyVault.ps1

```diff
@@ -0,0 +1,45 @@
+﻿param(
+    [Parameter(Mandatory)]
+    [string] $SubscriptionId,
+
+    [Parameter(Mandatory)]
+    [string] $KeyVaultName
+)
+
+Function Connect-AzureSubscription()
+{
+    # Ensure the user is logged in
+    try
+    {
+        $azureContext = az account show
+    }
+    catch {
+    }
+
+    if (-not $azureContext)
+    {
+        Write-Host "`nPlease login to Azure..."
+        az login
+        $azureContext = az account show
+    }
+
+    # Ensure the desired subscription is selected
+    if ($azureContext.id -ne $SubscriptionId)
+    {
+        Write-Host "`nSelecting subscription $SubscriptionId`n"
+        az account set --subscription $SubscriptionId
+    }
+
+    return $azureContext
+}
+
+Connect-AzureSubscription | Out-Null
+
+# Load all secrets from KeyVault and set the appropriate Environment Variable
+$ids = az keyvault secret list --subscription $SubscriptionId --vault-name $KeyVaultName --query '[*].id' --output tsv
+ForEach ($id in $ids){
+    $name = $id.split('/')[4].Replace('-', '_')
```

Comment:
When doing text extraction like this, it would be helpful to add a comment about the input received, and what part you are aiming to extract/transform.

Reply:
Done